### ISSUE TITLE
Discuss the next steps for Spring on Quarkus.

### ISSUE THREAD

**Issue by geoand** [snowdrop/project#64]
We have a wide selection of topics we can choose to work on:

- [ ] Implement SB AutoConfiguration mechanism
- [ ] Spring Cloud Stream
- [ ] Spring @Transactional
- [ ] Spring @Scheduled
- [ ] Spring Caching annotations
- [ ] Spring on Quarkus BOM
- [ ] Spring Test support
- [ ] Spring Batch
- [ ]  Move Spring Web to build time
- [ ] Spring Session
- [ ] Spring Data Mongo
- [ ] Spring Functions on Quarkus 
- [ ] Eureka support
- [ ] Support of Spring's ApplicationContext

**Comment by geoand** [comment]
Stuff that is already in progress:

* Spring Data REST (Gytis)
* Spring Jars (Aurea)
* Spring specific properties to quarkus properties conversion (Aurea)
* More Kubernetes support on Quarkus (Ioannis, Georgios)
* Spring Conditionals (Georgios)

**Comment by metacosm** [comment]
Why don't we create a SCL project instead of using this project?

**Comment by geoand** [comment]
I don't know what the best approach is. If someone has an opinion, let me
know and I'll do whatever, I don't mind

**Comment by geoand** [comment]
Spring Autoconfiguration, 
Spring Test, 
Spring Batch,
Spring build time (look into it with Bill)
Spring Cloud Stream (evaluate difficulty)


Eureka (Naina to evaluate if it makes sense)

**Comment by nainaz** [comment]
spring on Quarkus BOM: follow up with Quarkus team to know their thoughts.

**Comment by nainaz** [comment]
Spring @Transactional
Spring @Scheduled
Spring Caching annotations

How much efforts for them to figure where we can fit them delivery wise.

**Comment by geoand** [comment]
@geoand convert the decisions into a table where we show what we will try to do, what has the lowest priority, what we won't do etc.

**Comment by metacosm** [comment]
I thought we were going to use the quarkus fork to track quarkus-related issues?

**Comment by geoand** [comment]
We are, but since the overall picture is here for now I went ahead and broke them down

**Comment by cmoulliard** [comment]
Why is `quarkus eureka extension` moved to `evaluation` column @geoand  as an extension already exists - https://github.com/fmcejudo/quarkus-eureka  ? Having a configuration serve as eureka is more than a plus for cloud applications

**Comment by geoand** [comment]
Yeah, I wasn't really aware of that one. But even so it makes sense to evaluate that one and see if we want to help to make it better

**Comment by cmoulliard** [comment]
Do you know which Scheduler framework is used by Quarkus (Quartz, ...) ? @geoand
